client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -862,27 +992,49 @@ Response get() throws IOException {
      */
     public static class FailureListener {
         /**
-         * Notifies that the host provided as argument has just failed
+         * Notifies that the node provided as argument has just failed
          */
-        public void onFailure(HttpHost host) {
-
-        }
+        public void onFailure(Node node) {}
     }
 
     /**
-     * {@code HostTuple} enables the {@linkplain HttpHost}s and {@linkplain AuthCache} to be set together in a thread
-     * safe, volatile way.
+     * {@link NodeTuple} enables the {@linkplain Node}s and {@linkplain AuthCache}
+     * to be set together in a thread safe, volatile way.
      */
-    private static class HostTuple<T> {
-        final T hosts;
+    static class NodeTuple<T> {
+        final T nodes;
         final AuthCache authCache;
 
-        HostTuple(final T hosts, final AuthCache authCache) {
-            this.hosts = hosts;
+        NodeTuple(final T nodes, final AuthCache authCache) {
+            this.nodes = nodes;
             this.authCache = authCache;
         }
     }
 
+    /**
+     * Contains a reference to a blacklisted node and the time until it is
+     * revived. We use this so we can do a single pass over the blacklist.
+     */
+    private static class DeadNodeAndDeadness implements Comparable<DeadNodeAndDeadness> {
```

Comment:
nit: this is kind of a funny name to me, I would be ok with `DeadNode`, not a biggie though, it's a private class

Reply:
Will rename.